azavea/raster-foundry
# Add tile ingest

## Overview

This PR is the outcome of ADR 0012 and implements a structure for ingests that can be expanded upon as use-cases are introduced. Currently, file-based and s3-based inputs and outputs are supported.

### Notes

The jar produced through `app-server/ingest` takes two arguments:
`-j` - This parameter expects the URI for an ingest job specification. Such a structure can be found in `app-server/ingest/src/test/resources/`. `awsJob` runs an s3 input and s3 output ingest, whereas `localJob` runs a filesystem based input and output.
`-t` - A flag which tells the program that the ingest in question is a test and attempts to verify ingest upon completion, throwing an `Exception` in the case of missing layers.


## Testing Instructions

 * Run `scripts/cibuild` to build the ingest jar
 * Carry out a local ingest with `scripts/test`
 * A GeoTrellis catalog should be produced at `/tmp/landsat-rgb` per the `localJob` ingest specification

Supercedes #545 
Connects #587
